object-api-multipart.go

```diff
@@ -199,12 +199,6 @@ func (o objectAPI) ListMultipartUploads(bucket, prefix, keyMarker, uploadIDMarke
 		})
 	}
 	if uploadIDMarker != "" {
-		if strings.HasSuffix(keyMarker, slashSeparator) {
-			return result, probe.NewError(InvalidUploadIDKeyCombination{
-				UploadIDMarker: uploadIDMarker,
-				KeyMarker:      keyMarker,
-			})
-		}
 		id, e := uuid.Parse(uploadIDMarker)
```

Comment:
this check was valid, because if the marker is a directory (non-recursive) then uploadIDMarker has to be empty string.

uploadIDMarker is not empty only if the marker was a leaf.